python/ray/workflow/workflow_executor.py

```diff
@@ -266,50 +308,22 @@ async def _handle_ready_task(
                 f"[{workflow_id}@{task_id}]"
             )
 
-            # ---------------------- retry the task ----------------------
-            state.task_retries[task_id] += 1
-            total_retries = state.tasks[task_id].options.max_retries
-            if state.task_retries[task_id] <= total_retries:
-                logger.info(
-                    f"Retry [{workflow_id}@{task_id}] "
-                    f"({state.task_retries[task_id]}/{total_retries})"
-                )
-                state.construct_scheduling_plan(task_id)
+            if self._retry_failed_task(workflow_id, task_id, e):
```

Comment:
This is really confusing. `_retry_failed_task` actually do two things (check and retry). Maybe call it:

```
self._try_to_retry_failed_task(...)
```

Reply:
Also put comments about this and the next ones.

This is a big block checking different cases, so it's good that the reviewer can just read the code and don't need to jump to other functions to understand what's doing here.

Reply:
ah, I think breaking the giant function would be more clear. let me bring them back again

Reply:
@suquark  I think breaking it is ok. Just the naming is too confuse. Just rename to something better reflect the purpose is good.

Reply:
> This is a big block checking different cases, so it's good that the reviewer can just read the code and don't need to jump to other functions to understand what's doing here.

What I mean is that, when read the code (even it's a function), it's very clear about what we are doing there, then that's good enough.

Btw, we should keep the function small and scope it only doing one thing.